src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Gets the processor for the specified rule type.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\DateTimeProvider\CurrentDateTimeProvider;
+use \Automattic\WooCommerce\Admin\PluginsProvider\PluginsProvider;
+
+/**
+ * Class encapsulating getting the processor for a given rule type.
+ */
+class GetRuleProcessor {
+	/**
+	 * Get the processor for the specified rule type.
+	 *
+	 * @param string $rule_type The rule type.
+	 *
+	 * @return object The matching processor for the specified rule type, or a FailRuleProcessor if no matching processor is found.
+	 */
+	public static function get_processor( $rule_type ) {
+		switch ( $rule_type ) {
```

Comment:
Do you think it would make sense to have an interface or abstract class for the Rule Processors?

Reply:
Yes, this is something I would normally do. Not sure why I didn't do this originally. Done in 36488948.